pandas/core/computation/expr.py

```diff
@@ -39,8 +36,69 @@
 
 import pandas.io.formats.printing as printing
 
+# A token value Python's tokenizer probably will never use.
+_BACKTICK_QUOTED_STRING = 100
+
+Tok = Tuple[int, str]
+
+
+def _clean_special_characters_column_name(name: str) -> str:
+    """Check if name contains any special characters, if it contains any
+    special characters the special characters will be replaced by an special
+    string and an underscore suffix is added."""
+    if not isinstance(name, str):
+        return name
+
+    if name.isidentifier() and not iskeyword(name):
+        return name
+
+    # Create a list with the special characters and their replacement.
+    # So far we only replace single character operators.
+    special_characters_replacements = {
+        " ": "_",
+        **{
+            char: "_" + token.tok_name[tokval] + "_"
+            for char, tokval in tokenize.EXACT_TOKEN_TYPES.items()
+            if len(char) == 1
+        },
+    }
+
+    name = "".join(special_characters_replacements.get(char, char) for char in name)
+    name = "BACKTICK_QUOTED_STRING_" + name
+
+    # By my knowlegde the current conversion cannot fail, but who knows.
+    if not name.isidentifier():
+        raise ValueError(
+            "Could not convert %s to a valid Python identifier" % name +
+            "Use square bracket notation instead, "
+            "or raise an issue and see if we can implement and work around "
+            "for this situation as well."
+        )
+
+    return name
+
+
+def _tokenize_backtick_quoted_string(
+    token_generator: Generator[tokenize.TokenInfo, None, None]
```

Comment:
```suggestion
    token_generator: Iterable[tokenize.TokenInfo]
```

I think the Python docs suggest doing this when there is no send or return type for the `Generator` and would be more readable